secboot/secboot.go

```diff
@@ -251,3 +253,50 @@ func MarkSuccessful() error {
 
 	return nil
 }
+
+// AuthMode corresponds to an authentication mechanism.
+type AuthMode string
+
+const (
+	AuthModePassphrase AuthMode = "passphrase"
+	// TODO: Add PIN option when secboot support lands.
+)
+
+// VolumesAuthOptions contains options for the volumes authentication
+// mechanism (e.g. passphrase authentication).
+type VolumesAuthOptions struct {
+	Mode       AuthMode `json:"mode,omitempty"`
+	Passphrase string   `json:"passphrase,omitempty"`
+	// TODO: Add PIN option when secboot support lands.
+	KDFType string        `json:"kdf-type,omitempty"`
+	KDFTime time.Duration `json:"kdf-time,omitempty"`
```

Comment:
will client be given choice or should we rather provide sane defaults here? (or is the UI/installer doing that already?)

Reply:
kdf-type and kdf-time are optional, secboot uses sane defaults when empty values are passed. The client doesn't have to provide those.

Reply:
Looks like we reject empty defaults though?  https://github.com/canonical/snapd/pull/14845/files/05a4adfc69af1ac122e9b33370380e8aee26f0c1#diff-f2e78ed50950bc2284466f7345099d6ba7d0096a28f056db133a091e041ef044R291-R299

Reply:
Oh, I thought this happy test covered the optional bits, no? https://github.com/canonical/snapd/pull/14845/files/05a4adfc69af1ac122e9b33370380e8aee26f0c1#diff-42d6c1a4494e52c74cc6526991508b3edb2d8e00c9cffd3c1ab83d1838a0f258R3152-R3154